app/src/main/java/com/arctouch/publictransportation/fragment/DepartureListFragment.java

```diff
@@ -0,0 +1,170 @@
+package com.arctouch.publictransportation.fragment;
+
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.v4.app.Fragment;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.StaggeredGridLayoutManager;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+
+import com.android.volley.RequestQueue;
+import com.arctouch.publictransportation.R;
+import com.arctouch.publictransportation.domain.Departure;
+import com.arctouch.publictransportation.domain.Route;
+import com.arctouch.publictransportation.domain.generic.params.RouteIdParam;
+import com.arctouch.publictransportation.domain.generic.returns.DeparturesReturn;
+import com.arctouch.publictransportation.interfaces.VolleyListener;
+import com.arctouch.publictransportation.volley.VolleySingleton;
+import com.arctouch.publictransportation.volley.endpoints.DeparturesVolley;
+
+import java.util.List;
+
+public class DepartureListFragment extends Fragment implements VolleyListener<DeparturesReturn> {
+    public static final String ROUTE = "route";
+    public static final String SHOW_ROUTE_DETAILS = "show_route_details";
```

Comment:
As I mentioned before, you should create a Factory Method and make these two constants private, so DepartureListFragment becomes the only one responsible for managing its own dependencies. This method could look something like:
```
    public static DepartureListFragment create(@NonNull Route route, boolean showRouteDetails) {
        Bundle arguments = new Bundle();
        arguments.putSerializable(DepartureListFragment.ROUTE, route);
        arguments.putBoolean(DepartureListFragment.SHOW_ROUTE_DETAILS, showRouteDetails);

        DepartureListFragment fragment = new DepartureListFragment();
        fragment.setArguments(arguments);
        return fragment;
    }